base_rest/models/rest_service_registration.py

```diff
@@ -41,17 +41,40 @@ def _register_hook(self):
         # we have to to rebuild the registry. We use a new
         # registry so we have an empty cache and we'll add services in it.
         services_registry = self._init_global_registry()
+        self.build_registry(services_registry)
+
+    def build_registry(self, services_registry, states=None,
+                       exclude_addons=None):
+        if not states:
+            states = ('installed', 'to upgrade')
+        # lookup all the installed (or about to be) addons and generate
+        # the graph, so we can load the components following the order
```

Comment:
"we load REST, controllers following the order of the 'addons' dependencies to ensure that controllers defined in a more specialized addon and overriding more generic one takes precedences on the generic one into the registry"